client/omnichannel/directory/OmnichannelDirectoryPage.js

```diff
@@ -25,34 +28,60 @@ const OmnichannelDirectoryPage = () => {
 
 	const [contactReload, setContactReload] = useState();
 
+	const handleContactsVerticalBarCloseButtonClick = () => {
+		directoryRoute.push({});
+	};
+
+	const handleChatsVerticalBarCloseButtonClick = () => {
+		directoryRoute.push({ tab: 'chats' });
+	};
+
 	useEffect(() => {
 		if (!tab) {
 			return directoryRoute.replace({ tab: defaultTab });
 		}
 	}, [directoryRoute, tab, defaultTab]);
 
-	const ContactContextualBar = useCallback(() => {
+	const openInRoom = () => {
+		FlowRouter.go('live', { id });
+	};
+
+	const ContactContextualBar = () => <VerticalBar className={'contextual-bar'}>
+		<VerticalBar.Header>
+			{context === 'new' && <Box flexShrink={1} flexGrow={1} withTruncatedText mi='x8'><Icon name='user' size='x20' /> {t('New_Contact')}</Box>}
+			{context === 'info' && <Box flexShrink={1} flexGrow={1} withTruncatedText mi='x8'><Icon name='user' size='x20' /> {t('Contact_Profile')}</Box>}
+			{context === 'edit' && <Box flexShrink={1} flexGrow={1} withTruncatedText mi='x8'><Icon name='pencil' size='x20' /> {t('Edit_Contact_Profile')}</Box>}
+			<VerticalBar.Close onClick={handleContactsVerticalBarCloseButtonClick} />
+		</VerticalBar.Header>
+		{context === 'new' && <ContactNewEdit reload={contactReload} close={handleContactsVerticalBarCloseButtonClick} />}
+		{context === 'info' && <ContactInfo reload={contactReload} id={id} />}
+		{context === 'edit' && <ContactEditWithData id={id} reload={contactReload} close={handleContactsVerticalBarCloseButtonClick} />}
+	</VerticalBar>;
+
+	const ChatsContextualBar = () => <VerticalBar className={'contextual-bar'}>
+		<VerticalBar.Header>
+			{context === 'info' && <Box flexShrink={1} flexGrow={1} withTruncatedText mi='x8'><Icon name='info-circled' size='x20' /> {t('Room_Info')}</Box>}
+			{context === 'info' && <VerticalBar.Action title={t('View_full_conversation')} name={'new-window'} onClick={openInRoom} />}
+			<VerticalBar.Close onClick={handleChatsVerticalBarCloseButtonClick} />
+		</VerticalBar.Header>
+		{context === 'info' && <ChatInfo id={id} />}
+		{context === 'edit' && <EditRoomContextBar rid={id} />}
```

Comment:
We need a specific component / form to edit Omnichannel Rooms. The component called here is a global one.